scripts/add_directory.js

```diff
@@ -0,0 +1,287 @@
+    let resText = $("#batch-response");
+    let textClass, text;
+    if (suc == 0 && err == 0) {
+        text = "No batches added";
+        textClass = "text-info";
+    }
+    else if (err == 0) {
+        text = suc + " batches added";
+        textClass = 'text-success';
+    }
+    else if (suc == 0) {
+        text = err + " folders ignored, all already exist as batches";
+        textClass = 'text-warning';
+    }
+    else {
+        text = suc + " batches added, " + err + " ignored";
+    }
+    if (fadeTimeout) {
+        clearTimeout(fadeTimeout);
+        resText.css("display: none;");
+    }
+    resText[0].className = textClass;
+    resText.text(text);
+    resText.fadeIn("fast");
+    fadeTimeout = setTimeout(function() {
+        resText.fadeOut();
+        fadeTimeout = null;
+    }, 5000);
+}
+
+function createBatchUi(folder, num) {
+    return `<div data-batch="${num}" style="display: none;" class="batch panel panel-primary">
+            <div class="panel-heading">
+                <h3 id="batch-dir-${num}" class="panel-title">${folder}</h3>
+            </div>
+            <div class="panel-body">
+                <div class="row">
+                    <span class="col-xs-5">
+                        <div class="row">
+                            <span class="col-sm-4 col-xs-12">
+                                <label for="name-input-${num}">Batch Name</label>
+                            </span>
+                            <span class="col-sm-8 col-xs-12">
+                                <input id="name-input-${num}" type="text" placeholder="Batch name...">
+                            </span>
+                        </div>
+                    </span>
+                    <span class="col-xs-4">
+                        <input id="recursive-check-${num}" type="checkbox">
+                        <label for="recursive-check-${num}">Add recursively</label>
+                    </span>
+                    <span class="col-xs-3">
+                        <div class="row">
+                            <span class="col-sm-6 col-xs-12">
+                                <button id="batch-submit-${num}" class="btn btn-success" onclick="submitBatch(${num})"><span class="hidden rotating glyphicon glyphicon-repeat"></span>Create</button>
+                            </span>
+                            <span class="col-sm-6 col-xs-12">
+                                <button id="batch-cancel-${num}" class="btn btn-danger" onclick="deleteBatch(${num})">Cancel</button>
+                            </span>
+                        </div>
+                    </span>
+                </div>
+                <div class="alert-wrapper" id="alert-wrapper-${num}" class="row"></div>
+            </div>
+        </div>`;
+}
+
+function submitBatch(n) {
+    let batchReq = {
+        dir: $("#name-input-" + n).val(),
+        recursive: $("#recursive-check-" + n)[0].checked,
+        batch_name: $("#batch-dir-" + n).text()

```

Comment:
I think these should have default values just in case, maybe the directory name by default (especially if that will be displayed to the user in the batch selection screen)